src/opnsense/mvc/app/controllers/OPNsense/CheckIP/Api/SettingsController.php

```diff
@@ -0,0 +1,277 @@
+     * @param $nodeType
+     * @return result array
+     */
+    public function searchAction($nodeType = null)
+    {
+        $this->sessionClose();
+        if ($this->request->isPost() && $nodeType != null) {
+            $this->validateNodeType($nodeType);
+            $mdlCheckIPService = new Service();
+
+            $grid = new UIModelGrid($mdlCheckIPService->$nodeType);
+            $fields = array("default", "name", "url", "regex", "username", "password", "verifysslpeer", "description");
+            $grid = $grid->fetchBindRequest($this->request, $fields);
+
+            // Include the factory default check IP service.
+            $grid = $this->includeFDS($mdlCheckIPService, $grid);
+
+            return $grid;
+        }
+    }
+
+    /**
+     * validate nodeType
+     * @param $nodeType
+     * @throws \Exception
+     */
+    private function validateNodeType($nodeType = null)
+    {
+        if (array_search($nodeType, $this->nodeTypes) === false) {
+            throw new \Exception('unknown nodeType: ' . $nodeType);
+        }
+    }
+
+    /**
+     * is_default (is at least one CheckIP service enabled as the default; FDS inclusive)
+     * @return result array
+     */
+    public function is_defaultAction($nodeType = null)
+    {
+        $is_default = false;
+        if ($nodeType != null) {
+            $mdlCheckIPService = new Service();
+
+            $nodes = $mdlCheckIPService->$nodeType->getNodes();
+            foreach ($nodes as $nodeUuid => $node) {
+                $node = $mdlCheckIPService->getNodeByReference($nodeType . '.' . $nodeUuid);
+                $is_default = $node->default->__toString() == "1" ? true : $is_default;
+            }
+            $is_default = ($mdlCheckIPService->disable_factory_default_service->__toString() == "1") ? $is_default : true;
+        }
+        return array('result' => $is_default);
+    }
+
+    /**
+     * disable all CheckIP service items; FDS inclusive
+     * @param $nodeType
+     * @param &$mdlCheckIPService
+     */
+    private function disableAll(&$mdlCheckIPService, $nodeType)
+    {
+        $nodes = $mdlCheckIPService->$nodeType->getNodes();
+        foreach ($nodes as $nodeUuid => $node) {
+            $node = $mdlCheckIPService->getNodeByReference($nodeType . '.' . $nodeUuid);
+            $node->default = "0";
+        }
+        $mdlCheckIPService->disable_factory_default_service = "1";
+    }
+
+    /**
+     * Include the factory default check IP service.
+     */
+    private function includeFDS($mdlCheckIPService, $grid)
+    {
+        $mdlCheckIPFDS = new Factory_Default_Service();
+
+        $fds = $mdlCheckIPFDS->factory_default_service->getNodes();
+        $fds['default'] = ($mdlCheckIPService->disable_factory_default_service->__toString() == "1") ? "0" : "1";
+        $fds['uuid'] = 'FDS';
+
+        $grid['rows'][] = $fds;
+        $grid['rowCount'] = $grid['rowCount'] + 1;
+        $grid['total'] = $grid['total'] + 1;
```

Comment:
please note that this looks like the response expected by bootgrid.
This contains some data to query.